onyo/tests/test_usecases.py

```diff
@@ -0,0 +1,205 @@
+import subprocess
+import pytest
+from pathlib import Path
+
+from onyo.lib.onyo import OnyoRepo
+
+
+prefilled_assets = [["warehouse/monitor_dell_PH123.86JZho",
+                     "display: '22.0'\nfzj_inventory: '45FZ18'\n"],
+                    ["warehouse/monitor_dell_NoIdea.S0M3",
+                     "display: '27.0'\n"],
+                    ["warehouse/laptop_apple_macbook.oiw629",
+                     "RAM: '8GB'\ndisplay: '13.3'\nUSB_A: '2'\nfzj_inventory: '28FZ34'\nbuild-date: '20160501'\n"],
+                    ["warehouse/laptop_apple_macbook.9r32he",
+                     "RAM: '8GB'\ndisplay: '13.3'\nfzj_inventory: '28FJ34'\nbuild-date: '20180501'\n"],
+                    ["somegroup/userA/laptop_apple_macbook.9r5qlk",
+                     "RAM: '8GB'\ndisplay: '15.0'\nhostname: 'first.host'\n"],
+                    ["somegroup/userB/laptop_lenovo_thinkpad.owh8e2",
+                     "RAM: '8GB'\ndisplay: '14.6'\nfzj_inventory: '13BH9F'\n"],
+                    ["warehouse/laptop_lenovo_thinkpad.iu7h6d",
+                     "RAM: '8GB'\ndisplay: '13.3'\nfzj_inventory: '63AH90'\n"],
+                    ["warehouse/laptop_microsoft_surface.oq782j",
+                     "RAM: '8GB'\ndisplay: '12.4'\nfzj_inventory: '73CDA45'\ntouchscreen: 'yes'\n"],
+                    ]
+preset_dirs = ['retired', 'lost']
+
+
+@pytest.mark.repo_dirs(*preset_dirs)
+@pytest.mark.repo_contents(*prefilled_assets)
+def test_workflow_cli(repo: OnyoRepo) -> None:
+
+    # 1. create new group
+    workgroup = Path("newgroup")
+    cmd = ['onyo', 'mkdir', '--yes', '--quiet', str(workgroup)]
+    ret = subprocess.run(cmd, capture_output=True, text=True)
+    assert ret.returncode == 0
+
+    # 2. New member joining group
+    member = workgroup / "Sam User"
+
+    # 2a. Create the user
+    cmd = ['onyo', 'mkdir', '--yes', '--quiet', str(member)]
+    ret = subprocess.run(cmd, capture_output=True, text=True)
+    assert ret.returncode == 0
+
+    # 2b. Check the warehouse for a display
+    cmd = ['onyo', 'get', '-p', 'warehouse', '-H', '--filter', 'type=monitor', 'display=22.0']
```

Comment:
As a follow up, I notice that here in `'display=22.0'` the number is not quoted separately, so entered as an actual number, which is different from above. :)

Reply:
I was going to say: "Not really. A command line argument is always coming in as a string."
However, you appear to be right, in that `main:StoreKeyValuePairs` does indeed seem to convert it.
Will fix it.

Reply:
Switched to floats for display size everywhere (I hope) and pushed.